feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.ts

```diff
@@ -73,33 +116,27 @@ export class ImportEntriesDialogComponent {
     return form;
   }
 
-  selectFile(file: File, form: FormGroup) {
-    this.selectedFile = file;
-    this.importService.loadFile(file).subscribe(
-      (data) => {
-        if (this.importToCartService.isDataParsable(data)) {
-          this.fileError = {};
-          this.loadedFile = data as string[][];
-        } else {
-          this.fileError = { notParsable: true };
-          this.loadedFile = null;
-        }
-        form.get('file')?.updateValueAndValidity();
-      },
-      (error) => {
-        this.fileError = error;
-        this.loadedFile = null;
-        form.get('file')?.updateValueAndValidity();
-      }
-    );
+  protected validateMaxSize(file: File) {
+    if (
+      this.fileValidity?.maxSize &&
+      file.size / 1000000 > this.fileValidity?.maxSize
+    ) {
+      this.fileError.tooLarge = true;
+      throw Error();
+    }
   }
 
-  importProducts(): void {
-    if (this.loadedFile) {
-      this.importToCartService.loadProductsToCart(this.loadedFile, {
-        name: this.form.get('name')?.value,
-        description: this.form.get('description')?.value,
-      });
+  protected validateEmpty(data: string[][]) {
+    if (data.toString().length === 0) {
+      this.fileError.empty = true;
+      throw Error();
```

Comment:
What about adding some message for those thrown errors with  explanation what goes wrong?

Reply:
> What about adding some message for those thrown errors with explanation what goes wrong?

catch doesn't use the parameter because we have info already in this.fileError, throw is only to break pipe